magma/lib/magma/loader/record_entry.rb

```diff
@@ -125,13 +125,38 @@ def update_entry
 
     def temp_entry
       # Replace the entry with the appropriate values for the column.
-      @record.map do |att_name,value|
+      errors = []
+
+      real_id = @loader.real_id(@model, @record_name)
+      if real_id.nil?
+        errors.push("#{@model.model_name} '#{@record_name}' has no backing identifier, could not be loaded.")
```

Comment:
`real_id` can itself return nil in various cases as well, so we handle that explicitly.